cmd/controller/controller.go

```diff
@@ -19,15 +19,23 @@ import (
 var (
 	configPath string
 	masterURL  string
+	namespace  string
 )
 
+// Controller's own namespace is obtained here along with supported flags and env variables.
 // Note: kubeconfig hierarchy is 1) -kubeconfig flag, 2) $KUBECONFIG exported var. If neither is specified
 //   we'll do an in-cluster config, so for testing it's easiest to export KUBECONFIG.
 func init() {
+	const OWN_NAMESPACE = "OWN_NAMESPACE"
+	namespace = os.Getenv(OWN_NAMESPACE)
+	if namespace == "" {
```

Comment:
This will break if the controller is run out of cluster.  That's not problem if we only plan to run it in a pod.